### ISSUE TITLE
Suggestion: Drawdowns as % rather than $

### ISSUE THREAD

**Issue by ryankennedyio** [mhallsmoore/qstrader#90]
Drawdowns currently calculated by $. Keeping them as % would be a more realistic representation of performance, obviously as bigger $ swings with more capital may actually be a lower % drawdown -- but this isn't currently represented correctly.

Will require changing drawdown tests, manually calculating what the drawdown % should be at each of the 10 steps and slightly modifying implementation to do that.

**Comment by femtotrader** [comment]
It may be worth to share some code with https://github.com/quantopian/qrisk/

**Comment by femtotrader** [comment]
Some people also use pyfolio `https://github.com/mementum/backtrader/issues/108`

**Comment by nwillemse** [comment]
@femtotrader i have developed something similar to pyfolio for my own use on top of @mhallsmoore backtester from one of his books.  I want to develop something similar for qstrader and can contribute if you guys would like.  It looks like this:

![uis-v0 2](https://cloud.githubusercontent.com/assets/5043222/16954938/6d437ce6-4d98-11e6-964f-1a68afedab32.png)

**Comment by femtotrader** [comment]
That's very interesting @nwillemse 
Why not using directly pyfolio ? what was lacking ?
Any contribution is welcome!

**Comment by nwillemse** [comment]
I was trying to recreate my old Excel version of a "tearsheet".  A one pager that shows me all the information that i found important.  This way I can quickly compare the results of different systems.

I guess we can have a summary tearsheet and a more detailed backtest reports with all the different stats...

**Comment by ryankennedyio** [comment]
@nwillemse that looks fantastic actually.

Could that be implemented as another Statistics module?

Could it output that sheet _per strategy_ too ?

**Comment by mhallsmoore** [comment]
Hi all,

@nwillemse - This looks great! Very similar to the "one pagers" we used to receive in the institutional world. What are you using for the GUI? I can't tell if it is actually exported to Excel or whether you've built a nice GUI on top of Matplotlib.

I think it would be fantastic to have something like this in the system. Would you be willing to add code to the codebase for it?

Mike.

**Comment by femtotrader** [comment]
What is odd here is that there is for example "Trade Winning %"... 
it looks like more statistics for Forex than stock.
@nwillemse how do you consider that a trade is "winning" in qstrader because there is no bracket orders and I think you can consider that a trade is winning only after closing it (ie if it was a buy order, after selling same amount of shares).
Did you wrote a bracket orders bottler on top of qstrader ?

**Comment by nwillemse** [comment]
Hi everyone,

@mhallsmoore I used simply matplotlib and seaborn to output one composite figure made up of various subplots.  I have been looking at the qstrader code to see where I can add it.  I will look into the statistics module as @ryankennedyio mentioned.

However, I need some help in converting the list of executions into a list of trades.  For the Trade statistics @femtotrader I took the old SAT Backtester and output the positions as soon as they are closed.  Then used this to count winning trades/losing trades, %, etc to get all the trade specific stats.  What is the best way to implement something to keep track of positions?  A position will be all the orders to open the trade and all the orders to close the trade, then one position is created.

Nick

**Comment by ryankennedyio** [comment]
@nwillemse ,

Currently qstrader's `portfolio` class keeps track of every `position` it has ever made or closed. This is used to calculate equity value of the portfolio at each time period.

I think we can hit two birds with one stone here; https://github.com/mhallsmoore/qstrader/issues/91. 

That issue is caused by the `portfolio` class keeping track of closed positions. I think something like your proposed Statistics class is much better suited to keeping closed positions. 

I think the ideal solution will involve your extended Statistics class keeping record of all positions, and do an expensive calculation once, simply, at the end of the backtest (or whenever it is requested to output the graph) to graph all this. The `portfolio` class then doesn't need to worry about any of this, and only keeps track of positions that are still open, as well as the sum values of "realized_pnl" from all positions that were closed and sent to Statistics.

I'm just not sure at the moment how we can design the interface between these classes to be "pain free". Spaghetti should be avoided at all costs, and what I proposed could very quickly be spaghetti without being planned correctly.
